code/Control/UserDefinedFormController.php

```diff
@@ -262,25 +265,40 @@ public function process($data, $form)
             if (!empty($data[$field->Name])) {
                 if (in_array(EditableFileField::class, $field->getClassAncestry() ?? [])) {
                     if (!empty($_FILES[$field->Name]['name'])) {
-                        $foldername = $field->getFormField()->getFolderName();
-
-                        // create the file from post data
-                        $upload = Upload::create();
-                        try {
-                            $upload->loadIntoFile($_FILES[$field->Name], null, $foldername);
-                        } catch (ValidationException $e) {
-                            $validationResult = $e->getResult();
-                            foreach ($validationResult->getMessages() as $message) {
-                                $form->sessionMessage($message['message'], ValidationResult::TYPE_ERROR);
+                        $doReturn = false;
+
+                        $file = Versioned::withVersionedMode(function () use ($field, $form, $doReturn) {
+                            if (!$field->getFolderExists()) {
+                                $field->createProtectedFolder();
+                                $stage = Injector::inst()->get(self::class)->config()->get('file_upload_stage');
+                                Versioned::set_stage($stage);
```

Comment:
This needs to be outside the `if (!$field->getFolderExists()) {` block.  Otherwise if there is an existing folder, then the stage won't get set to draft before uploading files to it.

Reply:
I think in requirements was said that we set Draft stage only if folder is not exist. For all other cases we use current stage.